include/grpcpp/impl/codegen/server_callback.h

```diff
@@ -69,6 +69,31 @@ class ServerCallbackRpcController {
   // Allow the method handler to push out the initial metadata before
   // the response and status are ready
   virtual void SendInitialMetadata(std::function<void(bool)>) = 0;
+
+  /// SetCancelCallback passes in a callback to be called when the RPC is
+  /// canceled for whatever reason (streaming calls have OnCancel instead). This
+  /// is an advanced and uncommon use with several important restrictions.
+  ///
+  /// If code calls SetCancelCallback on an RPC, it must also call
+  /// ClearCancelCallback before calling Finish on the RPC controller.
+  ///
+  /// The callback should generally be lightweight and nonblocking and primarily
+  /// concerned with clearing application state related to the RPC or causing
+  /// operations (such as cancellations) to happen on dependent RPCs.
+  ///
+  /// If the RPC is already canceled at the time that SetCancelCallback is
+  /// called, the callback is invoked immediately.
+  ///
+  /// The cancellation callback may be executed concurrently with the method
+  /// handler that invokes it but will certainly not issue or execute after the
+  /// return of ClearCancelCallback.
+  ///
+  /// The callback is called under a lock that is also used for
+  /// ClearCancelCallback and ServerContext::IsCancelled, so the callback CANNOT
+  /// call either of those operations on this RPC or any other function that
+  /// causes those operations to be called before the callback completes.
+  virtual void SetCancelCallback(std::function<void()> callback) = 0;
+  virtual void ClearCancelCallback() = 0;
```

Comment:
Nit: Can you add a more explicit description of ClearCancelCallback and what it does?

Reply:
Will do